ambry-api/src/test/java/com.github.ambry/router/InMemoryRouter.java

```diff
@@ -102,6 +103,23 @@ public BlobProperties getBlobProperties() {
     public ByteBuffer getBlob() {
       return ByteBuffer.wrap(blob.array());
     }
+
+    /**
+     * @param range the {@link ByteRange} for the blob, or null.
+     * @return the blob content within the provided range, or the entire blob, if the range is null.
+     */
+    public ByteBuffer getBlobRange(ByteRange range) {
```

Comment:
the exception that this throws if the range is not resolvable will be caught by the caller, I will make this a bit more explicit by throwing a `RouterException` if the range is not satisfiable.